src/TeamSpeak3QueryApi/Query/Protocols/IProtocol.cs

```diff
@@ -0,0 +1,14 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace TeamSpeak3QueryApi.Net.Query.Protocols
+{
+    public interface IProtocol
+    {
+        Task<CancellationTokenSource> ConnectAsync();
+        CancellationTokenSource Connect(string username, string password);
```

Comment:
A `CancellationToken` should be returned, not a `CancellationTokenSource`

Reply:
Why? This function is not changed. You also returned the `CancellationTokenSource`. It´s needed for the processingloop. Did I miss something?